activeviam/atoti-ui-migration
# UI-8299 - Support notebooks in the Content Migration Tool

### Description

The PR is part of UI-8299. It does the following: 

- Add a migration script for Atoti 7.0 notebook, without handling for now the calculated measure (that's why I added an argument to `migrateWidget`.
- Refactor the `src/cli/bin.ts` to call the migration scripts depending on the file extension.
- Add some checks for the different Atoti UI / Atoti version in `bin.ts`.

What need to be done in follow-up PRs:

- [x] Handle case with calculated measures: as decided, we wouldn't handle this case as it is too much of an edge case.
- [x] Add a "migration notebooks" part in the readme: https://github.com/activeviam/atoti-ui-migration/pull/99

### Testing

- Create an Atoti notebook with Atoti 0.7, the AUI version used would be 5.0. Make sure it has a Atoti UI widget, with filters. I can share my test notebook if needed.
- Migrate the notebook with the command line:
 `npx -- activeui-migration -i test.ipynb -o migrated-notebook.ipynb -s servers.json --from-version 0.7 --to-version 0.8`
- Open the migrated notebook in JSON format and check that every widgetState in cells that represent widget have been updated.
